src/canvas.class.js

```diff
@@ -492,6 +492,20 @@
      * @return {Boolean}
      */
     isTargetTransparent: function (target, x, y) {
+      if (target.shouldCache && target._cacheCanvas && !target.isCacheDirty()) {
```

Comment:
the logic that tells us if the target is on its own cache canvas is a little bit more complex ( i have to verify )

Reply:
Ah ok. I've now noticed a canvas/path that this code doesn't properly handle, so I'm wondering if it's due to not checking its cache canvas properly or something.